src/v1/parser.rs

```diff
@@ -124,14 +124,99 @@ quick_error! {
 }
 
 /// Represents a type of the entry inside a signature file
-#[derive(Debug)]
+#[derive(Debug, PartialEq, Eq)]
 pub enum EntryKind<P: AsRef<Path>> {
     /// A directory
     Dir(P),
     /// A file or a symbolic link
     File(P),
 }
 
+impl<P: AsRef<Path>> EntryKind<P> {
+    /// Get path of the entry
+    pub fn path(&self) -> &Path {
+        use self::EntryKind::*;
+        match *self {
+            Dir(ref p) | File(ref p) => p.as_ref(),
+        }
+    }
+
+    /// Converts to EntryKind<&Path>
+    pub fn as_ref(&self) -> EntryKind<&Path> {
+        use self::EntryKind::*;
+        match *self {
+            Dir(ref p) => Dir(p.as_ref()),
+            File(ref p) => File(p.as_ref()),
+        }
+    }
+
+    /// Clones path and returns EntryKind<PathBuf>
+    pub fn cloned(&self) -> EntryKind<PathBuf> {
```

Comment:
`to_owned` is the method of the `ToOwned` trait. So I think we shouldn't use that name outside of `ToOwned` implementation.